src/index.ts

```diff
@@ -142,9 +146,23 @@ export function withSpid(
     samlConfig
   );
 
-  return loadSpidStrategyOptions()
+  return fromNullable(null)(appConfig.startupIdpsMetadata)
+    .map(parseStartupSpidStrategy)
+    .fold(
+      () => loadSpidStrategyOptions(),
+      idpOptionsRecord =>
+        new Task(() =>
```

Comment:
you can use https://github.com/gcanti/fp-ts/blob/1.12.0/src/Task.ts#L72 ie.

`Task.of(bindSpidStrategyOptions(..))`

prefer `of()` static factory to `new`

Reply:
Fixed in a4bdb13